File: GolangAns/Easy/Two_Sum.go
package main

import "fmt"

func main() {
	nums := []int{2, 7, 11, 15}
	twoSum(nums, 9)
	fmt.Println(twoSum(nums, 9))
}
func twoSum(nums []int, target int) []int {
	x, y := 0, 1
	for x < len(nums) {
		for ; y < len(nums); y++ {
			if nums[x]+nums[y] == target {
				return []int{x, y}
			}
		}
		x++
		y = x + 1
	}
	return nil
}

File: PythonAns/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for i, num in enumerate(nums):
            if target-num in d:
                return d[target-num], i
            d[num] = i


nums = [3, 2, 4]
i = Solution()
ans1, ans2 = i.twoSum(nums, 6)
print(ans1, ans2)
